Lib/test/test_httpservers.py

```diff
@@ -444,6 +444,46 @@ def test_head(self):
         self.assertEqual(response.getheader('content-type'),
                          'application/octet-stream')
 
+    def test_browser_cache(self):
+        #constructs the path relative to the root directory of the HTTPServer
+        response = self.request(self.base_url + '/test')
+        self.check_status_and_reason(response, HTTPStatus.OK, data=self.data)
+        last_modif = response.headers['Last-modified']
+
+        # send new request to the same url with request header 
+        # If-Modified-Since set to Last-Modified : must return 304
+        from email.message import Message
+        headers = Message()
+        headers['If-Modified-Since'] = last_modif
+        response = self.request(self.base_url + '/test', headers=headers)
+        self.check_status_and_reason(response, HTTPStatus.NOT_MODIFIED)
+
+        # if If-None-Match header is present, ignore If-Modified-Since
+        headers['If-None-Match'] = "*"
+        response = self.request(self.base_url + '/test', headers=headers)
+        self.check_status_and_reason(response, HTTPStatus.OK)        
+
+        # with If-Modified-Since earlier than Last-Modified, must return 200
+        import datetime
+        import email.utils
+        dt = email.utils.parsedate(last_modif)
+        # build datetime object : one year before last modification
+        old_dt = [dt[0] - 1] + list(dt[1:7])
+        old_dt = datetime.datetime(*old_dt)
+        headers = Message()
+        headers['If-Modified-Since'] = email.utils.format_datetime(old_dt)
+        response = self.request(self.base_url + '/test', headers=headers)
+        self.check_status_and_reason(response, HTTPStatus.OK)
+
+        # build datetime object : one hours after last modification
+        new_dt = [dt[0]] + [dt[1] + 1] + list(dt[2:7])
```

Comment:
Dt[1] is attribute tm_mon, but the comment suggests you want to adjust by one hour.